### PR TITLE
[cocoon]: sequential calls of encryption API result in nonce reuse (<=0.3.3)

### PR DESCRIPTION
**Pull request by fadeevab**

### CODE REVIEW

**Comment by tarcieri** [comment]
This is typically described as a nonce reuse vulnerability (see [CWE-323](https://cwe.mitre.org/data/definitions/323.html)).

Perhaps the title should be "sequential calls of encryption API result in nonce reuse" or "nonce reuse resulting in duplicate ciphertexts"

**Comment by fadeevab** [comment]
Fixed, check it out 👆

File: crates/cocoon/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+id = "RUSTSEC-0000-0000"
+package = "cocoon"
+date = "2023-10-15"
+url = "https://github.com/fadeevab/cocoon/issues/22"
+categories = ["crypto-failure"]
+cvss = "CVSS:3.1/AV:L/AC:H/PR:N/UI:N/S:C/C:L/I:L/A:N"
+keywords = ["nonce", "stream-cipher"]
+
+[affected]
+functions = {"cocoon::Cocoon::encrypt" = ["<= 0.3.3"], "cocoon::Cocoon::dump" = ["<= 0.3.3"], "cocoon::Cocoon::wrap" = ["<= 0.3.3"], "cocoon::MiniCocoon::encrypt" = ["<= 0.3.3"], "cocoon::MiniCocoon::dump" = ["<= 0.3.3"], "cocoon::MiniCocoon::wrap" = ["<= 0.3.3"]}
```
Comment (tarcieri) [inline crates/cocoon/RUSTSEC-0000-0000.md:12]:
```suggestion
[affected.functions]
"cocoon::Cocoon::encrypt" = ["<= 0.3.3"]
"cocoon::Cocoon::dump" = ["<= 0.3.3"]
"cocoon::Cocoon::wrap" = ["<= 0.3.3"]
"cocoon::MiniCocoon::encrypt" = ["<= 0.3.3"]
"cocoon::MiniCocoon::dump" = ["<= 0.3.3"]
"cocoon::MiniCocoon::wrap" = ["<= 0.3.3"]
```

**Comment by fadeevab** [comment]
Is it good now?

**Pull request merged**